projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -241,14 +244,16 @@ export class MultiSelectComponent<V> implements AfterContentInit, OnChanges {
       return;
     }
 
-    const selectedItems: SelectOptionComponent<V>[] | undefined = this.allOptionsList?.filter(item =>
-      this.isSelectedItem(item)
-    );
+    this.subscriptionLifecycle.add(
```

Comment:
It doesn't make sense to add a new subscription every time we set the trigger label. What's the bug?

Reply:
The bug is that the `allOptionsList` is not updating.

Reply:
Ever (this would surprise me)? Or that when options update, this logic isn't rerun (this I would expect).

Reply:
It's not updating from the call to ngOnChanges().

Reply:
Right, it wouldn't - it's not coming in as an input but as content children exposed via a QueryList. That will emit when they change (and that's what allOptions is projected from). The problem is this page is a mix of async and sync behavior - the set trigger behavior is updating regular variables, the search and options are working in observables.

My suggestion would be to represent the selections as observables too. If we keep the hybrid, we're stuck managing subscriptions which is easy to mess up.

Reply:
I really don't think it makes sense to refactor this component right now.

Reply:
@jake-bassett What property do you want it to change on? Why should options list change when an input changes?

Reply:
I could help if you could explain the use case

Reply:
AllOptionsList will update as soon as you add/remove any select option component. Are you trying to change the properties of select option component and expecting that it would reflect in multi select?

Reply:
Talked a bit offline. Removed the subscription. Please take a look when you get a chance.